src/libsyncengine/syncpal/syncpalworker.cpp

```diff
@@ -175,10 +175,12 @@ void SyncPalWorker::execute() {
                 stopAndWaitForExitOfAllWorkers(fsoWorkers, stepWorkers);
                 if ((stepWorkers[0] && workersExitCode[0] == ExitCode::SystemError &&
```

Comment:
It would be nice to have a carefully named function that computes this boolean value. The function would be easy to unit-test.